### ISSUE TITLE
Feature request: alert on different default parameter values in interface & implementation

### ISSUE THREAD

**Issue by BenMorel** [vimeo/psalm#7792]
Take the following example:

```php
interface I {
    public function test(?int $a = null): void;
}

class C implements I {
    public function test(?int $a = 1): void {
        var_export($a);
    }
}
```

Although this is valid PHP, [the interface lies](https://3v4l.org/MKpAa) about the actual default value in the implementation:

```php
function foo(I $i) {
    $i->test();
}

foo(new C); // 1
```

At the moment, [Psalm doesn't complain](https://psalm.dev/r/b44e97b691) here.

Would it be possible to implement an inspection for this?

**Comment by BenMorel** [comment]
Interesting opinion, I had not thought about it this way. What about the opposite, though?

```php
interface I {
    public function test(?int $a = 1): void;
}

class C implements I {
    public function test(?int $a = null): void {
    }
}
```

Or even:

```php
interface I {
    public function test(int $a = 1): void;
}

class C implements I {
    public function test(int $a = 2): void {
    }
}
```

**Comment by AndrolGenhald** [comment]
I think this might be better suited to a plugin. It could be useful in some cases, but I think I agree more with @orklah here. When I decide to call a method without an argument so that it uses the default, I see that as me saying "I want you to do your default thing, whatever that is" rather than "I see that your default is null and I want to call you with null so I'll just omit that argument".
